src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -354,7 +386,7 @@ protected Object invokeDescribable(String symbol, Object _args) {
 
         // The only time a closure is valid is when the resulting Describable is immediately executed via a meta-step
         NamedArgsAndClosure args = parseArgs(_args, metaStep!=null && metaStep.takesImplicitBlockArgument(),
-                UninstantiatedDescribable.ANONYMOUS_KEY, singleArgumentOnly);
```

Comment:
Hmm, I didn't notice this earlier. Does this mean that interpolation in describables is ignored? Or does it get detected once the step is invoked (probably not, since the `GString` will already have been flattened)? For example, try this as a test case:

```
archiveArtifacts("${PASSWORD}")
```

(The step is going to fail unless you have a password that happens to be a valid Ant pattern, but all we care about is whether the interpolation is detected)

Reply:
You're right, this won't get caught in  `flattenGString` because the `secretsDetector` variable is null. I'll  see about putting  an instance  of `InterpolatedSecretsDetector` in  `invokeDescribable`

Reply:
Instead of returning the `UninstantiantedDescribable` we now return the `NamedArgsAndClosure` which stores the interpolated strings.
See changes in 1afab594b3f0b0364a7bbed5aa6bbaf597bc19a0